test/built-ins/AsyncGeneratorFunction/for-await-of-interleaved.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/*---
+author: Maya Lekova <mslekova@chromium.org>
+esid: await
+description: >
+  for-await-of iteration and builtin Promises are properly interleaved,
+  meaning await in for-of loop takes only 1 tick on the microtask queue.
+flags: [async]
+features: [async-functions, async-iteration]
+---*/
+
+const actual = [];
+const expected = [
+  'Promise: 6',
+  'Promise: 5',
+  'Await: 3',
+  'Promise: 4',
+  'Promise: 3',
+  'Await: 2',
+  'Promise: 2',
+  'Promise: 1',
+  'Await: 1',
+  'Promise: 0'
+];
+const iterations = 3;
+
+async function* naturalNumbers(start) {
```

Comment:
Not sure why this function (and checkAssertions, countdown) are async.

Reply:
The `naturalNumbers` function is supposed to test awaiting on async generators (although it generates its data synchronously). If it's changed to a normal generator, the order of interleaving ("Promise" vs "Await") changes completely.
If `checkAssertions` function is not async, the possible error thrown from it gets swallowed up, even though the `$DONE` function is chained.
`countdown` indeed was unnecessary made async, so removed the keyword, thanks!